libs/cardano-ledger-binary/test-lib/Test/Cardano/Ledger/Binary/Twiddle.hs

```diff
@@ -0,0 +1,190 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstrainedClassMethods #-}
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Test.Cardano.Ledger.Binary.Twiddle
+  ( Twiddler (unTwiddler),
+    Twiddle (..),
+    encodingToTerm,
+    toTwiddler,
+    toTerm,
+  )
+where
```

Comment:
This is a huge file. Somebody should explain what's going on here.

Reply:
That's true, I should properly document this module. I'll create an issue for that.

Other than that, this looks fine to me, as the Twiddle instances should not care about the deserializer version unless we make some low level changes such as requiring all terms to be in canonical form or such.

Reply:
Here is a link to the ticket that tracks documenting the module: https://github.com/input-output-hk/cardano-ledger/issues/3067